### PR TITLE
Add GetBytes, GetChars and GetStream to SqliteDataReader

### PR DESCRIPTION
**Pull request by AlexanderTaeschner**
Addresses parts of #18

### CODE REVIEW

**Comment by bricelam** [comment]
FYI, we're locked down on features for version 2.1.0, so merging will have to wait until we get a 2.2.0 branch.

File: src/Microsoft.Data.Sqlite.Core/SqliteDataRecord.cs [hunk 1]
```diff
@@ -170,9 +171,20 @@ public static Type GetFieldType(string type)
...
         public virtual long GetChars(int ordinal, long dataOffset, char[] buffer, int bufferOffset, int length)
-            => throw new NotSupportedException();
+        {
+            var text = GetString(ordinal);
+            text.CopyTo((int)dataOffset, buffer, bufferOffset, length);
+            return length;
+        }
+
+        public virtual Stream GetStream(int ordinal)
+            => new MemoryStream(GetBlob(ordinal));
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite.Core/SqliteDataRecord.cs]: Make it read-only.

Comment (bricelam) [inline src/Microsoft.Data.Sqlite.Core/SqliteDataRecord.cs]: I wonder if multiple calls should share the same buffer instance.

Comment (AlexanderTaeschner) [reply src/Microsoft.Data.Sqlite.Core/SqliteDataRecord.cs]: Done.

Comment (AlexanderTaeschner) [reply src/Microsoft.Data.Sqlite.Core/SqliteDataRecord.cs]: Done.

**Pull request merged**